### PR TITLE
Fix ClassNotFoundException for Spring Boot 2.3.0

### PR DESCRIPTION
**Pull request by matsumana**
Hi, I tried Spring Boot 2.3.0 GA with Armeria 0.99.5
But my app couldn't start due to ClassNotFoundException.

HealthStatusHttpMapper has been deprecated since 2.2.0 and deleted since 2.3.0
We can use `HttpCodeStatusMapper` or `SimpleHttpCodeStatusMapper` instead of it.

```
2020-05-16 00:54:21.586  WARN 88826 --- [           main] o.s.boot.SpringApplication               : Unable to close ApplicationContext

java.lang.IllegalStateException: Failed to introspect Class [com.linecorp.armeria.spring.actuate.ArmeriaSpringActuatorAutoConfiguration] from ClassLoader [org.springframework.boot.loader.LaunchedURLClassLoader@5010be6]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:481) ~[spring-core-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:358) ~[spring-core-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:414) ~[spring-core-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.lambda$getTypeForFactoryMethod$2(AbstractAutowireCapableBeanFactory.java:743) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708) ~[na:na]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getTypeForFactoryMethod(AbstractAutowireCapableBeanFactory.java:742) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:681) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:649) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1608) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doGetBeanNamesForType(DefaultListableBeanFactory.java:526) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanNamesForType(DefaultListableBeanFactory.java:497) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansOfType(DefaultListableBeanFactory.java:619) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeansOfType(DefaultListableBeanFactory.java:611) ~[spring-beans-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.getBeansOfType(AbstractApplicationContext.java:1242) ~[spring-context-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	at org.springframework.boot.SpringApplication.getExitCodeFromMappedException(SpringApplication.java:880) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.getExitCodeFromException(SpringApplication.java:868) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.handleExitCode(SpringApplication.java:855) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.handleRunFailure(SpringApplication.java:806) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:325) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.0.RELEASE.jar!/:2.3.0.RELEASE]
	at info.matsumana.psystrike.Application.main(Application.java:16) ~[classes!/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:564) ~[na:na]
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49) ~[psystrike-psystrike-1.0.3-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:109) ~[psystrike-psystrike-1.0.3-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58) ~[psystrike-psystrike-1.0.3-SNAPSHOT.jar:na]
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:88) ~[psystrike-psystrike-1.0.3-SNAPSHOT.jar:na]
Caused by: java.lang.NoClassDefFoundError: org/springframework/boot/actuate/health/HealthStatusHttpMapper
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method) ~[na:na]
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3244) ~[na:na]
	at java.base/java.lang.Class.getDeclaredMethods(Class.java:2387) ~[na:na]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:463) ~[spring-core-5.2.6.RELEASE.jar!/:5.2.6.RELEASE]
	... 29 common frames omitted
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.actuate.health.HealthStatusHttpMapper
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:435) ~[na:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[na:na]
	at org.springframework.boot.loader.LaunchedURLClassLoader.loadClass(LaunchedURLClassLoader.java:129) ~[psystrike-psystrike-1.0.3-SNAPSHOT.jar:na]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[na:na]
	... 33 common frames omitted
```

Spring Boot sources:

- https://github.com/spring-projects/spring-boot/blob/v2.2.7.RELEASE/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/HealthStatusHttpMapper.java#L31-L32
- https://github.com/spring-projects/spring-boot/blob/v2.2.7.RELEASE/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/HttpCodeStatusMapper.java
- https://github.com/spring-projects/spring-boot/blob/v2.2.7.RELEASE/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/health/SimpleHttpCodeStatusMapper.java

### CODE REVIEW

**Comment by ikhoon** [comment]
Bye the way, how about upgrading to Spring Boot 2.3.0?

**Comment by matsumana** [comment]
> Bye the way, how about upgrading to Spring Boot 2.3.0?

I'll work on it as another PR.

**Comment by anuraaga** [comment]
This new class was added in spring boot 2.2, so this change might break compatibility with 2.0 and 2.1. Can we check with spring team whether it's intended to remove a class on a minor version change which shouldn't be allowed?

**Comment by anuraaga** [comment]
Though I guess they'll just say they do sometimes delete on minor versions :( Let's copy the logic in instead of using the class directly?

**Comment by matsumana** [comment]
I needed to care about the compatibility as well, sorry.
I'll copy it.

File: spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/SimpleHttpCodeStatusMapper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            final String code = getUniformCode(entry.getKey());
+            if (code != null) {
+                result.putIfAbsent(code, entry.getValue());
+            }
+        }
+        return Collections.unmodifiableMap(result);
+    }
+
+    @Nullable
+    private static String getUniformCode(@Nullable String code) {
```
Comment (anuraaga) [inline spring/boot-actuator-autoconfigure/src/main/java/com/linecorp/armeria/spring/actuate/SimpleHttpCodeStatusMapper.java:62]: Thanks, looking at it I think this class has a lot of complexity due to being user-configurable with string properties in Spring, but we don't need such configuration. How about we just declare the `Map` and use it directly without all this other stuff? I don't think we will ever need this normalization, etc.

**Comment by trustin** [comment]
`HttpCodeStatusMapper` (since 2.2.0) and `HealthStatusHttpMapper` (since 2.0.0) are exposed in the public API so that a user can override the default behavior. Are we removing that extension point?

**Comment by anuraaga** [comment]
I'd remove it, there doesn't seem like a clean way without separating out armeria-spring versions for spring boot minor version. Seems like overkill for a rarely used knob.

**Comment by adriancole** [comment]
second the remove. we release often, so if someone says there's a regression, we'll hear quickly then can re-visit complexity of it

**Comment by matsumana** [comment]
updated

**Comment by matsumana** [comment]
Thank you for merging!

**Pull request merged**